node_modules/oae-zencoder/lib/util.js

```diff
@@ -0,0 +1,129 @@
+/*!
+ * Copyright 2016 Apereo Foundation (AF) Licensed under the
+ * Educational Community License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License. You may
+ * obtain a copy of the License at
+ *
+ *     http://opensource.org/licenses/ECL-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an "AS IS"
+ * BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+var _ = require('underscore');
+var util = require('util');
+var Zencoder = require('zencoder');
+
+var log = require('oae-logger').logger('oae-zencoder');
+var PreviewConstants = require('oae-preview-processor/lib/constants');
+var PreviewUtil = require('oae-preview-processor/lib/util');
+var TenantsAPI = require('oae-tenants/lib/api');
+var ContentConfig = require('oae-config').config('oae-content');
+
+var ZencoderConfig = require('oae-config').config('oae-zencoder');
+
+var config = null;
+
+/**
+ * Set the default config. This can be overridden for individual tenants
+ *
+ * @return {Object} Zencoder configuration
+ */
+var init = module.exports.init = function(_config) {
+    config = _config;
+};
+
+/**
+ * Get the Zencoder API values that have been configured in the Admin UI.
+ *
+ * @return {Object} The Zencoder configuration for the tenant or the generic config for the host
+ */
+var getConfig = module.exports.getConfig = function(tenantAlias) {
+    var apiKey = ZencoderConfig.getValue(tenantAlias, 'zencoder', 'apiKey');
+
+    if (apiKey) {
+        return _.extend({}, config, {
+            'apiKey': apiKey,
+            'watermark': null,
+            'amazons3-region': ZencoderConfig.getValue(tenantAlias, 'zencoder', 'amazons3-region') || config['amazons3-region'],
+            'amazons3-bucket': ZencoderConfig.getValue(tenantAlias, 'zencoder', 'amazons3-bucket') || config['amazons3-bucket'],
+            'amazons3-access-key': ZencoderConfig.getValue(tenantAlias, 'zencoder', 'amazons3-access-key') || config['amazons3-access-key'],
+            'amazons3-secret-key': ZencoderConfig.getValue(tenantAlias, 'zencoder', 'amazons3-secret-key') || config['amazons3-secret-key']
```

Comment:
Here we're kind of doing a merge of tenant configuration values with system configuration values. Did you consider strictly using one or the other based on if the tenant is enabled? I'm not sure which approach is better, but here's a scenario:

a) A tenant puts in their API key, region, bucket, access key but not the secret key (so, it's their public key and our secret key)
b) They invoke some requests that ultimately cause request failures against their account
c) Is there potentially some way for them to see am audit log of what secret key was used against their amazon account?

Further more, adding default values "under the hood" (i.e., they can't see the values we're applying here, not even in the admin ui) could make configuration issues a bit trickier to solve for them.